Philips.CodeAnalysis.MaintainabilityAnalyzers/Naming/NamespaceMatchAssemblyNameAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Naming
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class NamespaceMatchAssemblyNameAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = @"Namespace matches Assembly Name";
+		private const string MessageFormat = @"Namespace and Assembly Name must match";
+		private const string Description = @"In order to prevent pollution of namespaces, and maintainability of namespaces, the Assembly Name and Namespace must match.";
+		private const string Category = Categories.Naming;
+
+		private static readonly DiagnosticDescriptor Rule = new(Helper.ToDiagnosticId(DiagnosticId.NamespaceMatchAssemblyName), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
```

Comment:
Consider trying the new SingleDiagnosticAnalyzer (partly to confirm it's good-to-go before I start migrating others to it)

Reply:
Done, looks good-to-go for me